sycl/source/detail/kernel_bundle_impl.hpp

```diff
@@ -442,7 +442,11 @@ class kernel_bundle_impl {
     return SetInDevImg || MSpecConstValues.count(std::string{SpecName}) != 0;
   }
 
-  const device_image_plain *begin() const { return &MDeviceImages.front(); }
+  const device_image_plain *begin() const {
+    assert(!MDeviceImages.empty() && "MDeviceImages can't be empty");
```

Comment:
Could you please clarify why MDeviceImages can't be empty?
I believe this should behave as `std::vector` which has `end() == begin()` if `empty()` is `true`.

Reply:
Agree that this is not a valid assert. I'll submit a fix as a separate pull request.

Reply:
`MDeviceImage`s can't be empty because `MDeviceImages.front()` is UB in case of `MDeviceImages.empty() == true`.

Reply:
@dm-vodopyanov it's UB to access `front`, but it doesn't mean, that kernel_bundle must have any device image at all. The spec mentions `empty()` member function for `kernel_bundle`, which @romanovvlad refers to: https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_the_kernel_bundle_class